tarofreighter/chain_porter.go

```diff
@@ -502,39 +468,66 @@ func (p *ChainPorter) storeAndTransferProofs(pkg *sendPackage) {
 	return
 }
 
-// advanceStateUntil advances the state machine up to, but not including, the
-// target state.
-func (p *ChainPorter) advanceStateUntil(currentPkg *sendPackage,
-	txBroadcastRespChan chan *PendingParcel,
-	targetState SendState) (*sendPackage, error) {
+// advanceState advances the state machine.
+func (p *ChainPorter) advanceState(currentPkg *sendPackage,
+	txBroadcastRespChan chan *PendingParcel) error {
 
-	log.Infof("ChainPorter advancing from state=%v to state=%v",
-		currentPkg.SendState, targetState)
+	logState := func(sendState SendState) {
+		log.Infof("ChainPorter advancing from state: %v", sendState)
+	}
 
-	continueStateProgression := currentPkg.SendState < targetState
+	// Advance through sync evaluated states. (Up to, but not including,
+	// SendStateWaitTxConf.)
+	continueStateProgression := currentPkg.SendState < SendStateWaitTxConf
 	for continueStateProgression {
+		logState(currentPkg.SendState)
+
 		// Before we attempt a state transition, make sure that we
 		// aren't trying to shut down.
 		select {
 		case <-p.Quit:
-			return nil, fmt.Errorf("porter shutting down")
+			return fmt.Errorf("porter shutting down")
 
 		default:
 		}
 
 		updatedPkg, err := p.stateStep(*currentPkg, txBroadcastRespChan)
 		if err != nil {
-			return nil, err
+			return err
 		}
 
 		// Continue state progression whilst target state has not yet
 		// been reached. Stop before evaluating target state.
-		continueStateProgression = updatedPkg.SendState < targetState
+		continueStateProgression = updatedPkg.SendState < SendStateWaitTxConf
```

Comment:
Perhaps let's create a like `SendStateTerminal`, which is always the last defined state in the iota? Then we just need to make sure all other states are added before this one, then we can base things off this abstract terminal state.